src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -895,6 +940,9 @@ else if (!getGuild().getSelfMember().hasPermission(getTextChannel(), Permission.
     @Override
     public RestAction<Message> crosspost()
     {
+        if (isEphemeral())
+            throw new IllegalStateException("Cannot suppress embeds on ephemeral messages.");
```

Comment:
Is this documented?

Reply:
I don't know but i tried to suppress embeds by retrieving the message and suppressing embeds, it says unknown message

```java
event.reply("https://www.youtube.com/watch?v=dQw4w9WgXcQ")
				.setEphemeral(true)
				.queue(interactionHook -> interactionHook.retrieveOriginal().queue(m -> {
					m.suppressEmbeds(true).queue();
				}));
```

Reply:
Crosspost does say that it may throw UNKNOWN_MESSAGE if the message is ephemeral